planner/core/logical_plan_builder.go

```diff
@@ -2194,9 +2192,11 @@ func (b *PlanBuilder) buildDataSource(tn *ast.TableName) (LogicalPlan, error) {
 // BuildDataSourceFromView is used to build LogicalPlan from view
 func (b *PlanBuilder) BuildDataSourceFromView(dbName model.CIStr, tableInfo *model.TableInfo) (LogicalPlan, error) {
 	charset, collation := b.ctx.GetSessionVars().GetCharsetInfo()
+	sqlMode := b.ctx.GetSessionVars().SQLMode
+
 	viewParser := parser.New()
 	viewParser.EnableWindowFunc(b.ctx.GetSessionVars().EnableWindowFunction)
-	selectNode, err := viewParser.ParseOneStmt(tableInfo.View.SelectStmt, charset, collation)
+	selectNode, err := viewParser.ParseOneStmt(tableInfo.View.SelectStmtBySQLMode(sqlMode), charset, collation)
```

Comment:
We can force viewParser to use non-ANSIQuote mode by `viewParser.SetSQLMode(sqlMode & ~mysql.ModeANSIQuotes)`

Reply:
I don't think this will work. is parser handle two different quote when `JOIN` tables, something wrong would happened.

Reply:
Do you mean that
```sql
select * from v1, "t2"; -> select * from (select * from `t1`), "t2";
```

Since we just merge the ast instead of insert the string, I don't think this will cause something wrong. And after lexer, we do not have any difference on the two types of quotes.